scripts/generate_activity_script/src/script.ts

```diff
@@ -346,29 +325,492 @@ async function withdrawRandomNFT(
    */
   const NFTID: string =
     withdrawnNFTs[Math.floor(Math.random() * withdrawnNFTs.length)];
-  try {
-    const res = await sendNFT(
-      withdrawalVoucher,
-      withdrawalManager,
-      address,
-      NFTID
-    );
-    const { code, hash } = await neutronWallet.clientCW.getTx(res.txHash);
-    if (code !== 0) {
+
+  const res = await sendNFT(
+    withdrawalVoucher,
+    withdrawalManager,
+    address,
+    NFTID
+  );
+  const { code, hash } = await neutronWallet.clientCW.getTx(res.txHash);
+  if (code !== 0) {
+    throw `Check up given hash ${hash}`;
+  }
+  return res;
+}
+
+/* Function dedicated to do IBC transfer
+ * This function used by two other functions in this script:
+ *  - IBCToTransfer
+ *  - IBCFromTransfer
+ * We've to use signAndBroadcast instead of sendIbcTokens since second is deprecated
+ */
+async function IBCTransfer(
+  clientSG: SigningStargateClient,
+  addressFrom: string,
+  addressTo: string,
+  channel: string,
+  port: string,
+  amount: Coin,
+  fee: Coin
+): Promise<string> {
+  const { transactionHash } = await clientSG.signAndBroadcast(
+    addressFrom,
+    [
+      {
+        typeUrl: "/ibc.applications.transfer.v1.MsgTransfer",
+        value: {
+          sourcePort: port,
+          sourceChannel: channel,
+          token: amount,
+          sender: addressFrom,
+          receiver: addressTo,
+          timeoutHeight: "0",
+          timeoutTimestamp: String(
+            (Math.floor(Date.now() / 1000) + 10 * 60) * 1e9
+          ),
+        },
+      },
+    ],
+    {
+      gas: "200000",
+      amount: [fee],
+    },
+    ""
+  );
+
+  return transactionHash;
+}
+
+/* Function dedicated to do IBC transfer from Neutron to Remote Chain
+ * Function doesn't handle any exceptions from the inner call
+ */
+async function IBCToTransfer(
```

Comment:
kinda strange name for the function

Reply:
Well, I covered the origin for this name in README, but if you have a better suggestion then I'll change it